lib/system.nim

```diff
@@ -4210,6 +4210,23 @@ when hasAlloc or defined(nimscript):
       x[j+i] = item[j]
       inc(j)
 
+  func insert*(x: var string, c: char, i = 0.Natural) =
+    ## Inserts character `c` into `x` at position `i`
+    ##
+    ## .. code-block:: Nim
+    ##   var a = "hllo"
+    ##   a.insert('e', 1) # a <- "Hello"
+    var xl = x.len
+    setLen(x, xl+1)
+    var j = xl-1
+    while j >= i:
+      shallowCopy(x[j+1], x[j])
+      dec(j)
+    j = 0
+    while j < 1:
+      x[j+i] = c
+      inc(j)
```

Comment:
Is this loop any different than just writing `x[i] = c`?

Reply:
Of course - I wasn't thinking, I just copied the insert(string, string) implementation and changed it to use a char instead. Will fix!